app/decorators/controllers/solidus_gdpr/spree/api/users_controller_decorator.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module SolidusGdpr
+  module Spree
+    module Api
+      module UsersControllerDecorator
+        def emails
+          users_scope = model_class.accessible_by(current_ability, :read)
+          orders_scope = ::Spree::Order.all
+
+          users_emails = users_scope.ransack(params[:q]).result.pluck(:email)
+          orders_emails = orders_scope.ransack(params[:q]).result(distinct: true).pluck(:email)
```

Comment:
We can't do it this way, this can load potentially millions of strings in memory. You should probably use `UNION` instead. The good news is you can still leverage the SQL generated by ActiveRecord by doing something such as:

```ruby
users_sql = users_scope.ransack(params[:q]).result.select(:email)
orders_sql = orders_scope.ransack(params[:q]).result(distinct: true).select(:email)

result = ActiveRecord::Base.connection.execute <<~SQL
  SELECT emails.* FROM (#{users_sql} UNION #{orders_sql}) emails
SQL
```